applications/HDF5Application/custom_io/hdf5_file.h

```diff
@@ -265,6 +265,28 @@ class KRATOS_API(HDF5_APPLICATION) File
         const TDataType& rData,
         WriteInfo& rInfo);
 
+    /**
+     * @brief Write a dataset to the hDF5 file.
+     *
+     * Performs collective write in MPI. The data is written blockwise according to
+     * processor rank.
+     *
+     * @tparam TDataType            Data type of the provided data.
+     * @tparam TIntegerType         Integer type for the shape.
+     * @param rPath                 Path to which the data is written.
+     * @param pData                 Pointer to the data.
+     * @param pShapeIteratorBegin   Iterator to the begin of the shape.
+     * @param pShapeIteratorEnd     Iterator to the end of the shape.
+     * @param rInfo                 Information about the written data (output).
+     */
+    template<class TDataType, class TIntegerType>
+    void WriteDataSet(
+        const std::string& rPath,
+        TDataType const * pData,
+        TIntegerType const * pShapeIteratorBegin,
+        TIntegerType const * pShapeIteratorEnd,
```

Comment:
I know what you mean but the name is super confusing because it contains both `p` and `Iterator`. This makes me think that you expect an array of pointers pointing to iterators marking the `begin`s of some other arrays (i.e.: they represent a nested array).

This goes for the rest of the new functions too.

```suggestion
        TIntegerType const * itShapeBegin,
        TIntegerType const * itShapeEnd,
```

Reply:
done :)